src/video_core/buffer_cache/buffer_cache.cpp

```diff
@@ -689,6 +830,109 @@ bool BufferCache::SynchronizeBufferFromImage(Buffer& buffer, VAddr device_addr,
     return true;
 }
 
+void BufferCache::SynchronizeRange(VAddr device_addr, u32 size) {
+    if (device_addr == 0) {
+        return;
+    }
+    ForEachBufferInRange(device_addr, size, [&](BufferId buffer_id, Buffer& buffer) {
+        if (buffer.is_deleted) {
```

Comment:
A buffer picked from ForEachBufferInRange cannot be in a deleted state, so this check is unnecessary